src/Panel/Panel.php

```diff
@@ -109,32 +602,260 @@ public static function render(App $kirby)
+        $routes[] = [
+            'pattern' => 'plugins/(:any)',
+            'action'  => function ($id) {
+                return [
+                    'component' => 'k-plugin-view',
+                    'icon' => 'box',
+                    'breadcrumbLabel' => $id,
+                    'props' => [
+                        'id' => $id
+                    ]
+                ];
+            },
+        ];
+        // @codeCoverageIgnoreEnd
+
+        // catch all route
+        $routes[] = [
+            'pattern' => '(:all)',
+            'action'  => function () use ($kirby) {
+                return 'The view could not be found';
+            }
+        ];
+
+        return $routes;
+    }
+
+
+    /**
+     * Set the current language in multi-lang
+     * installations based on the session or the
+     * query language query parameter
+     *
+     * @param \Kirby\Cms\App $kirby
+     * @return string|null
+     */
+    public static function setLanguage(App $kirby): ?string
+    {
+        // language switcher
+        if ($kirby->options('languages')) {
+            $session  = $kirby->session();
+            $language = get('language') ?? $session->get('panel.language', 'en');
+
+            // keep the language for the next visit
+            $session->set('panel.language', $language);
+
+            // activate the current language in Kirby
+            $kirby->setCurrentLanguage($language);
+
+            return $language;
+        }
+
+        return null;
+    }
+
+    /**
+     * Set the currently active Panel translation
+     * based on the current user or config
+     *
+     * @param \Kirby\Cms\App $kirby
+     * @return string
+     */
+    public static function setTranslation(App $kirby): string
+    {
+        if ($user = $kirby->user()) {
+            // use the user language for the default translation
+            $translation = $user->language();
+        } else {
+            // fall back to the language from the config
+            $translation = $kirby->panelLanguage();
+        }
+
+        $kirby->setCurrentTranslation($translation);
+
+        return $translation;
+    }
+
+    /**
+     * Returns data array for view
+     *
+     * @param \Kirby\Cms\App $kirby
+     * @param array $area
+     * @param array $view
+     * @return array
+     */
+    public static function view(App $kirby, ?array $area = null, array $view = []): array
+    {
+        // merge view with area defaults
+        $view = array_replace_recursive($area ?? [], $view);
+
+        $view['breadcrumb'] = $view['breadcrumb'] ?? [];
+        $view['path']       = Str::after($kirby->path(), '/');
+        $view['props']      = $view['props'] ?? [];
+        $view['search']     = $view['search'] ?? $kirby->option('panel.search.type', 'pages');
+
+        // make sure that routes are gone
+        unset($view['routes']);
 
-        return new Response($view->render());
+        // resolve lazy props
+        return $view;
```

Comment:
Was `A::apply()` used here before? That's what the comment suggests. If so, we can just remove the comment.

Reply:
I believe so, must be a leftover. Let's remove it.